remotes/handlers.go

```diff
@@ -139,6 +140,12 @@ func Fetch(ctx context.Context, ingester content.Ingester, fetcher Fetcher, desc
 		return err
 	}
 
+	if len(desc.Data) > 0 {
```

Comment:
On second look, this check isn't necessary, right? If Size is zero and data is too, we're fine, right?